src/engine/pieces/piece.ts

```diff
@@ -1,12 +1,25 @@
 import Player from '../player';
 import Board from '../board';
-import Square from '../square';
+import Square, { SquareState } from '../square';
+import PositionChange from '../positionChange';
+
+export enum PieceType {
+    KING,
+    QUEEN,
+    ROOK,
+    BISHOP,
+    KNIGHT,
+    PAWN
+}
```

Comment:
```suggestion
    KING = "King",
```
You want to explicity define the values of your enums like the above as by default they have the values 0,1,2... etc

and this can cause issues
- as the first element has the value 0 which will evaluate to false e.g. if you check if(PieceType.King) and the others will evaluate to true
- if we are sending these values in API requests they'll be read as 0,1,2... which would be confusing